sqtom/fitting_1d.py

```diff
@@ -26,18 +26,55 @@
 
 
 import numpy as np
+from scipy.optimize import root_scalar
 from lmfit import Minimizer, Parameters
 from sqtom.forward_solver import degenerate_pmf
 
 
+def two_schmidt_mode_guess(pd_data, sq_label="sq_"):
+    """Given a single mode histogram, this function generates a "physically" motivated guess for the loss, Schmidt occupations
+    and dark counts parameters.
+
+    This model is sensible only if the average g2 is above 2.
+
+    Args:
+        pd_data (array): rectangular array with the probability mass functions of the photon events
+        sq_label (string): label for the squeezing parameters.
+
+    Returns:
+        dict: dictionary containing a set of "reasonable" model parameters
+    """
+    res = marginal_calcs_1d(pd_data)
+    nmean = res["n"]
+    g2 = res["g2"]
+    P0 = pd_data[0]
+
+    def findeta(eta, nmean, g2, P0):
+        a = nmean / 4
+        b = (3 - g2) * (nmean ** 2) / 4 - nmean
+        c = (g2 - 3) * nmean ** 2
+        d = (3 - g2) * nmean ** 2 + 2 * nmean + 1 - 1 / P0 ** 2
+        return a * eta ** 3 + b * eta ** 2 + c * eta + d
+
+    eta = root_scalar(findeta, args=(nmean, g2, P0), x0=0, x1=1).root
+    n0 = (nmean + np.sqrt((g2 - 2) * nmean ** 2 - eta * nmean)) / (2 * eta)
+    n1 = (nmean - np.sqrt((g2 - 2) * nmean ** 2 - eta * nmean)) / (2 * eta)
+    noise = nmean / 1000
```

Comment:
I would sugest making this fraction, an optional argument in the guess, i.e., changing the signature of the function to be ` two_schmidt_mode_guess(pd_data, sq_label="sq_", noise_fraction = 0.001)` and the use this variable to do `noise = nmean * noise_fraction`

Reply:
Implemented 😄